src/contracts/core/Slasher.sol

```diff
@@ -5,233 +5,475 @@ import "./SlasherStorage.sol";
+    /// operator => allocator
+    /// could perhaps use 
+    mapping(address => address) public allocatorFor;
+
     constructor(
         IStrategyManager _strategyManager,
-        IDelegationManager _delegationManager,
-        IOperatorSetManager _operatorSetManager
-    ) SlasherStorage(_strategyManager, _delegationManager, _operatorSetManager) {}
+        IDelegationManager _delegationManager
+    ) SlasherStorage(_strategyManager, _delegationManager) {}
 
     /**
-     * @notice Called by an AVS to increase its own slashing request for a given
-     * operator set and operator in the current epoch
+     * @notice Queues slashing magnitude deallocation for an 
+     * (operator, strategy, operatorSet) tuple.
+     * Queues magnitude deallocation 21 days from now which will be automatically applied
+     * at that respective time.
+     * 
+     * @param operator the operator whom the slashing parameters are being changed
+     * @param deallocationParams deallocation params with differences to subtract from current magnitudes
+     * @param allocatorSignature if non-empty is the signature of the allocator on 
+     * the modification. if empty, the msg.sender must be the allocator for the 
+     * operator
      *
-     * @param operator the operator that the calling AVS is to increase the bips they want to slash
-     * @param operatorSetID the id of the operator set the AVS is increasing their slashing for
-     * @param strategies the list of strategies slashing requested is being modified for
-     * @param bipsToIncrease the basis points slashing to modify for given strategies
+     * @dev pushes a MagnitudeUpdate and TotalAndNonslashableUpdate to take effect in 21 days
+     * @dev reverts if magnitudeDiff > allocated magnitude
+     * since you cannot deallocate more than is already allocated
+     * @dev emits event queueDeallocation
      */
-    function increaseRequestedBipsToSlash(
+    function queueDeallocation(
         address operator,
-        bytes4 operatorSetID,
-        IStrategy[] memory strategies,
-        uint32 bipsToIncrease
-    ) external {
-        require(bipsToIncrease > 0, "Slasher.increaseRequestedBipsToSlash: bipsToIncrease must be positive");
-        require(
-            bipsToIncrease < SlashingAccountingUtils.BIPS_FACTOR,
-            "Slasher.increaseRequestedBipsToSlash: bipsToIncrease must be less than BIPS_FACTOR"
-        );
-        _modifyRequestedBipsToSlash(
-            operator, operatorSetID, strategies, EpochUtils.currentEpoch(), int32(bipsToIncrease)
-        );
+        MagnitudeAdjustmentParam[] calldata deallocationParams,
+        SignatureWithExpiry calldata allocatorSignature // TODO: implement signature verification
+    ) external returns(uint32 effectTimestamp) {
+        // perform allocator signature verification if not allocator or operator
+        if (msg.sender != operator && msg.sender != getAllocatorFor(operator)) {
```

Comment:
needs to be allocator

Reply:
operator can only call it if they are also allcoator

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/623/commits/f7c9a75affd75db460bf53ff9dc6d5b8ff9d32db
Will add signature verification later